data/src/main/kotlin/com/vultisig/wallet/data/repositories/TokenPriceRepository.kt

```diff
@@ -178,25 +183,79 @@ internal class TokenPriceRepositoryImpl @Inject constructor(
 
     private suspend fun fetchPricesWithContractAddress(
         chain: Chain,
-        tokens: List<Coin>,
+        contractAddresses: List<String>,
         currencies: List<String>,
-    ): Map<String, CurrencyToPrice> =
-        if (chain == Chain.Solana) emptyMap()
-        else coinGeckoApi.getContractsPrice(
-            chain = chain,
-            contractAddresses = tokens.map { it.contractAddress },
-            currencies = currencies,
+    ): Map<String, CurrencyToPrice> {
+        return if (chain == Chain.Solana) emptyMap()
+        else {
+            coroutineScope {
+                val coinGeckoContractsPrice = coinGeckoApi.getContractsPrice(
+                    chain = chain,
+                    contractAddresses = contractAddresses,
+                    currencies = currencies,
+                )
+                val notInCoinGeckoTokens = contractAddresses.filterNot { address ->
+                    coinGeckoContractsPrice.keys.any { key ->
+                        key.equals(
+                            address,
+                            false
+                        )
+                    }
+                }
+
+                notInCoinGeckoTokens.takeIf { it.isNotEmpty() }
+                    ?: return@coroutineScope coinGeckoContractsPrice
+
+                val tetherPrice = fetchTetherPrice()
+                val currency = currencies.first()
+                val lifiContractsPrice = notInCoinGeckoTokens
+                    .map { contractAddress ->
+                        async {
+                            contractAddress to getLifiContractPriceInUsd(
+                                chain,
+                                contractAddress
+                            )
+                        }
+                    }.awaitAll().associate { (contractAddress, priceInUsd) ->
+                        contractAddress to mapOf(currency to priceInUsd * tetherPrice)
```

Comment:
Since Lifi provides prices in USD, we use USDT to convert them into the local currency